src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1012,13 +1012,21 @@ private function addCacheSection(ArrayNodeDefinition $rootNode)
                             ->prototype('array')
                                 ->fixXmlConfig('adapter')
                                 ->beforeNormalization()
-                                    ->ifTrue(function ($v) { return (isset($v['adapters']) || \is_array($v['adapter'] ?? null)) && isset($v['provider']); })
+                                    ->ifTrue(function ($v) { return \is_array($v['adapter'] ?? null) && isset($v['provider']); })
                                     ->thenInvalid('Pool cannot have a "provider" while "adapter" is set to a map')
                                 ->end()
+                                ->beforeNormalization()
+                                    ->ifTrue(function ($v) { return \is_array($v['adapters'] ?? null) && 1 < \count($v['adapters']) && isset($v['provider']); })
+                                    ->thenInvalid('Pool cannot have a "provider" while "adapters" contains more than one value')
```

Comment:
Let's merge both case with a common error:
`Pool cannot have a "provider" while more than on adapter is defined.`